packager/media/codecs/es_descriptor.cc

```diff
@@ -167,13 +167,17 @@ void DecoderConfigDescriptor::WriteInternal(BufferWriter* writer) {
   writer->AppendNBytes(buffer_size_db_, 3);
   writer->AppendInt(max_bitrate_);
   writer->AppendInt(avg_bitrate_);
-  decoder_specific_info_descriptor_.Write(writer);
+
+  if (!decoder_specific_info_descriptor_is_ommited_)
```

Comment:
It does not seem that the flag is necessary. What about updating the descriptor to skip if it is empty instead, i.e.

```c++
if (!decoder_specific_info_descriptor_.data().empty())
  decoder_specific_info_descriptor_.Write(writer);

...

  if (decoder_specific_info_descriptor_.data().empty())
    return data_size_without_children;
```